### PR TITLE
Replace function declarations with arrow function expressions in examples

### PR DESCRIPTION
**Pull request by Kiarokh**
# ⚠️ Note
This PR is only for examples in the documentations.
I have not touched the components' code for replacing any function declarations with arrow function expressions!

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

**Comment by Kiarokh** [comment]
I did not touch the examples of `limel-file`, because they are already fixed here: https://github.com/Lundalogik/lime-elements/pull/1277

**Comment by Kiarokh** [comment]
I did not change the examples of `limel-snackbar` because I didn't understand how to change them to arrow functions

**Comment by Kiarokh** [comment]
There are some `constructor()`s which I have left untouched in the example.
I didn't know how to convert them to arrow functions.
For example:
````tsx
constructor() {
        this.checkValidity = this.checkValidity.bind(this);
    }
````
In input-field examples.

There are more of these individual ones left. But I can't fix them by my own :(

File: src/components/icon-button/icon-button.tsx [hunk 1]
```diff
@@ -63,11 +58,11 @@ export class IconButton {
         }
     }
 
-    private removeFocusedStyleOnClick() {
+    private removeFocusedStyleOnClick = () => {
```
Comment (Befkadu1) [inline src/components/icon-button/icon-button.tsx:61]: Put this change in other commit and the type of change is not `docs`. It should be `refactor`

File: src/components/picker/examples/picker-multiple.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { ListItem } from '@limetech/lime-elements';
 import { Component, h, State } from '@stencil/core';
 
 /**
- * Multiple values
+ * Multiple values can be picked
```
Comment (Befkadu1) [inline src/components/picker/examples/picker-multiple.tsx:5]: You think the change in the header looks better?

File: src/components/picker/examples/picker.tsx [hunk 3]
```diff
@@ -2,7 +2,9 @@ import { ListItem } from '@limetech/lime-elements';
 import { Component, h, State } from '@stencil/core';
 
 const NETWORK_DELAY = 500;
-
+/**
+ * Single value can be picked
+ */
```
Comment (Befkadu1) [inline src/components/picker/examples/picker.tsx:7]: What about this one `Single value` ? The sentence doesn't look like a heading

File: src/components/picker/examples/picker-multiple.tsx (hunk 2 above)
Comment (Kiarokh) [reply src/components/picker/examples/picker-multiple.tsx:5]: Yes, I think the new header explains the example more clearly.

File: src/components/picker/examples/picker.tsx (hunk 3 above)
Comment (Kiarokh) [reply src/components/picker/examples/picker.tsx:7]: I think the new header explains the example more clearly.

File: src/components/icon-button/icon-button.tsx (hunk 1 above)
Comment (Kiarokh) [reply src/components/icon-button/icon-button.tsx:61]: this seems not to be from the examples, so this was done by mistake. :/ I don't want to change anything in the components' code in this PR

Comment (Befkadu1) [reply src/components/icon-button/icon-button.tsx:61]: ok then you remove this change, right?

**Comment by Befkadu1** [comment]
> ```tsx
> constructor() {
>         this.selectedItem = this.items.filter((item: ListItem) => {
>             return !!item.selected;
>         })[0];
>     }
> ```
> 
> in `limel-list`
> 
> There are more of these individual ones left. But I can't fix them by my own :(

You don't have to fix it. Here the `constructor` used for initializing the local state of the component by assigning an object to `selectedItem` state. You can leave it like this.

File: src/components/icon-button/icon-button.tsx (hunk 1 above)
Comment (Kiarokh) [reply src/components/icon-button/icon-button.tsx:61]: I'll do

**Comment by Befkadu1** [comment]
> There are some `constructor()`s which I have left untouched in the example.
> I didn't know how to convert them to arrow functions.
> For example:
> 
> ```tsx
> constructor() {
>         this.checkValidity = this.checkValidity.bind(this);
>     }
> ```
> 
> In input-field examples.
> 

You can remove the constructor and change the method keyword `private` to `protected` like this

```
 @Watch('required')
    protected checkValidity() {
       ....
    }
```
    

This keyword allows a little more access than private members but a lot less than the public.

File: src/components/icon-button/icon-button.tsx (hunk 1 above)
Comment (Kiarokh) [reply src/components/icon-button/icon-button.tsx]: pushed a fixup

File: src/components/icon-button/icon-button.tsx [hunk 4]
```diff
@@ -19,7 +19,6 @@ export class IconButton {
         this.removeFocusedStyleOnClick =
             this.removeFocusedStyleOnClick.bind(this);
     }
-
```
Comment (Befkadu1) [inline src/components/icon-button/icon-button.tsx:22]: Remove this change

**Review by Befkadu1 — COMMENTED** [review]
The commit message "chore(chip-set): replace `chipSetOnChange` with `handleChange` following best practices & consistency" has more than 100 characters. There is a lint error in the build

File: src/components/icon-button/icon-button.tsx (hunk 4 above)
Comment (Kiarokh) [reply src/components/icon-button/icon-button.tsx:22]: yes sir

**Review by Befkadu1 — APPROVED** [review]
@adrianschmidt this PR is fine for me. You can look a bit and merge it if you are ok with the changes.

File: src/components/checkbox/examples/checkbox.tsx [hunk 5]
```diff
@@ -32,41 +25,49 @@ export class CheckboxExample {
...
-                        onChange={this.changeHandler}
+                        onChange={this.handleChange}
                     />
+                </div>
+                <p>
                     <limel-flex-container justify="end">
-                        <limel-button
-                            onClick={this.onClickToggleEnabled}
-                            label={this.disabled ? 'Enable' : 'Disable'}
+                        <limel-checkbox
```
Comment (adrianschmidt) [inline src/components/checkbox/examples/checkbox.tsx:33]: Haha! This is pretty meta 😄

Comment (Kiarokh) [reply src/components/checkbox/examples/checkbox.tsx:33]: 😝 for consistency. At some point of time, I want to make all these "settings" sections in examples look the same, and be visually separated from the component itself.

Comment (jgroth) [reply src/components/checkbox/examples/checkbox.tsx:33]: The "Composite" examples are all quite similar and quite easy to implement
https://github.com/Lundalogik/lime-elements/blob/main/src/components/button-group/examples/button-group-composite.tsx

Some of them can get quite complex however (not in the implementation but the settings UI :smile: )
https://github.com/Lundalogik/lime-elements/blob/main/src/components/menu/examples/menu-composite.tsx
https://github.com/Lundalogik/lime-elements/blob/main/src/components/chip-set/examples/chip-set-composite.tsx

Comment (adrianschmidt) [reply src/components/checkbox/examples/checkbox.tsx:33]: I'm thinking maybe we can build a component that uses the schema, but can build and display a nicer form for a number of common and "known" properties, and then just show any "left over" properties underneath, just like we do in the composite examples…? 🤔

Comment (jgroth) [reply src/components/checkbox/examples/checkbox.tsx:33]: Yeah sure, that would be possible :slightly_smiling_face: We could just modify the schema a bit before we use it in the form, hiding those settings in a collapsible section

Comment (Kiarokh) [reply src/components/checkbox/examples/checkbox.tsx:33]: sounds great

Comment (adrianschmidt) [reply src/components/checkbox/examples/checkbox.tsx:33]: Feels like a potential Hack-Tuesday… 🙂

Comment (Kiarokh) [reply src/components/checkbox/examples/checkbox.tsx:33]: 👩‍💻🧑‍💻👨‍💻

**Pull request merged**